docs/src/tutorials/linear/diet.jl

```diff
@@ -91,30 +91,21 @@ limits = DataFrames.DataFrame(
 
 model = Model(HiGHS.Optimizer)
 
-# Next, we create a set of decision variables `x`, indexed over the foods in the
-# `data` DataFrame. Each `x` has a lower bound of `0`.
+# Next, we create a set of decision variables `x`, with one element for each row
+# in the DataFrame. Each `x` has a lower bound of `0`, and we store the vector
+# as a new column `x` in the DataFrame `foods`:
 
-@variable(model, x[foods.name] >= 0);
+foods.x = @variable(model, x[1:size(foods, 1)] >= 0);
 
-# Our objective is to minimize the total cost of purchasing food. We can write
-# that as a sum over the rows in `data`.
+# Our objective is to minimize the total cost of purchasing food:
 
-@objective(
-    model,
-    Min,
-    sum(food["cost"] * x[food["name"]] for food in eachrow(foods)),
-);
+@objective(model, Min, sum(foods.cost .* foods.x));
 
 # For the next component, we need to add a constraint that our total intake of
-# each component is within the limits contained in the `limits` DataFrame.
-# To make this more readable, we introduce a JuMP `@expression`
-
-for limit in eachrow(limits)
-    intake = @expression(
-        model,
-        sum(food[limit["name"]] * x[food["name"]] for food in eachrow(foods)),
-    )
-    @constraint(model, limit.min <= intake <= limit.max)
+# each component is within the limits contained in the `limits` DataFrame:
+
+for row in eachrow(limits)
+    @constraint(model, row.min <= sum(foods[!, row.name] .* foods.x) <= row.max)
```

Comment:
I would suggest changing the `:name` column in `limits` to `:nutrient`, then updating `row.name` to `row.nutrient`. Otherwise, it makes it more challenging to understand that there are two different sets of things labelled `name`.